ui/app/AppLayouts/Wallet/panels/ActivityFilterPanel.qml

```diff
@@ -0,0 +1,264 @@
+        }
+    }
+
+    Repeater {
+        model: root.assetsList
+        delegate: ActivityFilterTagItem {
+            tagPrimaryLabel.text: symbol
+            iconAsset.icon: Style.png("tokens/" + symbol)
+            iconAsset.color: "transparent"
+            visible: !allChecked && checked
+            onClosed: root.toggleToken(symbol)
+        }
+    }
+
+    Repeater {
+        model: root.collectiblesList
+        delegate: ActivityFilterTagItem {
+            tagPrimaryLabel.text: name
+            iconAsset.icon: model.iconSource
+            iconAsset.color: "transparent"
+            visible: !allChecked && checked
+            onClosed: root.toggleCollectibles(name)
+        }
+    }
+
+    Repeater {
+        model: root.recentsList
+        delegate: ActivityFilterTagItem {
+            property int transactionType: to.toLowerCase() === root.store.overview.mixedcaseAddress.toLowerCase() ? Constants.TransactionType.Receive : Constants.TransactionType.Send
+            tagPrimaryLabel.text: transactionType === Constants.TransactionType.Receive ?
+                                      root.store.getNameForAddress(from) || StatusQUtils.Utils.elideText(from,6,4) :
+                                      root.store.getNameForAddress(to) || StatusQUtils.Utils.elideText(to,6,4)
+            visible: !allChecked && checked
+            onClosed: root.toggleRecents(transactionType === Constants.TransactionType.Receive ? from : to)
+        }
+    }
+
+    Repeater {
+        model: root.savedAddressList
+        delegate: ActivityFilterTagItem {
+            tagPrimaryLabel.text: ens.length > 0 ? ens : chainShortNames + StatusQUtils.Utils.elideText(address,6,4)
+            visible: !allChecked && checked
+            onClosed: root.toggleSavedAddress(address)
+        }
+    }
+
+    ActivityFilterMenu {
+        id: activityFilterMenu
+
+        selectedTime: root.selectedTime
+        onSetSelectedTime: {
+            if(selectedTime === Constants.TransactionTimePeriod.Custom) {
+                dialog.open()
+            }
+            else
+                root.changeSelectedTime(selectedTime)
+        }
+
+        typeFilters: root.typeFilters
+        onUpdateTypeFilter: root.toggleType(type)
+
+        statusFilters: root.statusFilters
+        onUpdateStatusFilter: root.toggleStatus(status)
+
+        tokensList: root.assetsList
+        collectiblesList: root.collectiblesList
+        onUpdateTokensFilter: root.toggleToken(tokenSymbol)
+        onUpdateCollectiblesFilter: root.toggleCollectibles(name)
+
+        store: root.store
+        recentsList: root.recentsList
+        savedAddressList: root.savedAddressList
+        onUpdateSavedAddressFilter: root.toggleSavedAddress(address)
+        onUpdateRecentsFilter: root.toggleRecents(address)
+    }
+
+    StatusDateRangePicker {
+        id: dialog
+        anchors.centerIn: parent
+        destroyOnClose: false
+        // To-do sync with backend
```

Comment:
still todo?

Reply:
Yes next PR will include the glue with backend